### DISCUSSION TITLE
Relationships to JPL Rules and formal guidelines

### DISCUSSION THREAD

**Discussion by dwagmuse** [NASA-AMMOS/slim#158] [General]
Love your recommended best practices! Seems like there ought to be some reference to governing rules because there are rules governing how much process is necessary for different classes of software. It's a little frustrating that there are different JPL sites offering different advice and guidance on different parts of the problem.   
https://pal.jpl.nasa.gov/about-sqi/  offers such advice but the advice isn't entirely consistent.
https://software.jpl.nasa.gov/   seems like a larger advocacy site you might want to link from.

The best practices seem to assume that you already have an established/funded project with requirements, etc. We have many software products that have no consistent supporting project or at least insufficient funding to support all of the best practices. Might be helpful to discuss how to tune the best practices for, e.g., one-(fractional)-person teams because that's the reality.

Thanks

**Comment by riverma** [comment]
> Love your recommended best practices! Seems like there ought to be some reference to governing rules because there are rules governing how much process is necessary for different classes of software. It's a little frustrating that there are different JPL sites offering different advice and guidance on different parts of the problem. https://pal.jpl.nasa.gov/about-sqi/ offers such advice but the advice isn't entirely consistent. https://software.jpl.nasa.gov/ seems like a larger advocacy site you might want to link from.
> 
> The best practices seem to assume that you already have an established/funded project with requirements, etc. We have many software products that have no consistent supporting project or at least insufficient funding to support all of the best practices. Might be helpful to discuss how to tune the best practices for, e.g., one-(fractional)-person teams because that's the reality.
> 
> Thanks

@dwagmuse - thank you so much for reviewing our best practices and for sharing extra links to institutional recommendations! I'll review those links more deeply - we've in the past done outreach to leads in the software strategy arena. Seems like a great time to touch back in with the institutional folks thinking along similar lines. 

> Might be helpful to discuss how to tune the best practices for, e.g., one-(fractional)-person teams because that's the reality.

That's a great point David. One way we try to address this is through our different tiers of community dissemination of best practices. For example, we have a tier where people can just read the guides and apply on their own to their project. We also have a tier where projects can join as _community members_, where the SLIM project can help push out best practices to the project itself. See our [Join page](https://nasa-ammos.github.io/slim/docs/join/) for more details.
